pkg/validator/sarif.go

```diff
@@ -0,0 +1,54 @@
+package validator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type SarifValidator struct{}
+
+func (SarifValidator) Validate(b []byte) (bool, error) {
+	var report map[string]interface{}
+	err := json.Unmarshal(b, &report)
+	if err != nil {
+		customErr := getCustomErr(b, err)
+		return false, customErr
+	}
+
+	schemaURL, ok := report["$schema"]
+	if !ok {
+		return false, errors.New("error - no schema found")
+	}
+
+	if _, ok := schemaURL.(string); !ok {
+		return false, errors.New("error - schema isn't a string")
+	}
+
+	loadedSchema := gojsonschema.NewReferenceLoader(schemaURL.(string))
```

Comment:
This will fail in airgapped environments or corporate networks without a proxy. I think we need a separate error if it can't download the schema from the URL. The current error is ` error: error - schema isn't valid: https://docs.oasis-open.org/sarif/sarif/v2.1.0/errata01/os/schemas/sarif-schema-2.1.0.json` which is confusing

Reply:
That's a great point. 
What do you think if we add an additional function that will verify that the schema's hostname is resolvable and try to establish a TCP connection prior to loading it?

If it's not resolvable or the connection can't be established, we might return with an error that indicates that schema couldn't be loaded.

Reply:
>What do you think if we add an additional function that will verify that the schema's hostname is resolvable and try to establish a TCP connection prior to loading it? If it's not resolvable or the connection can't be established, we might return with an error that indicates that schema couldn't be loaded.

I like that approach

Reply:
Added initial implementation for it, please take a look when you have a chance @kehoecj